distribution/test/itests/test-itests-common/src/test/java/ddf/common/test/restito/ChunkedContent.java

```diff
@@ -110,9 +158,28 @@ public static Action chunkedContentWithHeaders(String responseMessage, Duration
          */
         private ChunkedContentFunction(String responseMessage, Duration messageDelay,
                 int numberOfFailures) {
-            this.responseMessage = responseMessage;
+            this.responseMessage = responseMessage.toCharArray();
+            this.messageDelayMs = messageDelay.toMillis();
+            this.numberOfFailures = numberOfFailures;
+            this.getRequestHeaders = (Void) -> null;
+        }
+        /**
+         * Constructor for a response that has a delay and a number of planned failures.
+         * Supports range headers.
+         *
+         * @param responseMessage  Message to be sent in response.
+         * @param messageDelay     Time to wait between sending each character of the message.
+         * @param numberOfFailures Number of times to fail (simulate network disconnect) after sending
+         *                         the first character. Once this number is reached, the message will
+         *                         send successfully.
+         * @param getRequestHeaders Function that returns headers on the incoming request.
+         */
+        private ChunkedContentFunction(String responseMessage, Duration messageDelay,
+                int numberOfFailures, Function<Void, Map<String, String>> getRequestHeaders) {
```

Comment:
Instead of taking in a `Function`, couldn't we just take in a `HeaderCapture requestHeaders` (assuming it's now a top-level class, see my comment below)?. If that doesn't work for some reason, we should use a `Supplier` instead of `Function`, that way we don't have to use `Void` and `apply(null)` on L198.

Also, since those are related to the request, maybe the request headers should be the first argument?

With all those parameters, maybe we should take the time to apply the builder pattern to this class? Especially since we don't have too many people using it right now.